src/components/timeline/TrackContextMenu.js

```diff
@@ -586,32 +676,105 @@ class TimelineTrackContextMenuImpl extends PureComponent<Props> {
     );
   }
 
+  renderTrackSearchField() {
+    const { rightClickedTrack } = this.props;
+    const { searchFilter } = this.state;
+    if (rightClickedTrack !== null) {
+      // This option should only be visible in the top context menu and not when
+      // user right clicks.
+      return null;
+    }
+
+    return (
+      <React.Fragment>
+        <TrackSearchField
+          className="trackContextMenuSearchField"
+          currentSearchString={searchFilter}
+          onSearch={this._onSearch}
+          ref={this._trackSearchFieldElem}
+        />
+        <div className="react-contextmenu-separator" />
+      </React.Fragment>
+    );
+  }
+
+  _onSearch = (value: string) => {
+    this.setState({ searchFilter: value });
+  };
+
+  _onShow = () => {
+    const trackFieldElement = this._trackSearchFieldElem.current;
+    if (
+      // We need to focus the track search filter. But we can't use autoFocus
+      // property because this context menu is already rendered and hidden during
+      // the load of the web page. So, when user clicks on the track context menu
+      // button, focus will be moved to that component and search filter will not
+      // be focused anymore. To fix this, we are manually calling the focus.
```

Comment:
In React, the `autoFocus` property works when the component is mounted. As you correctly write, the context menu is mounted at load time, so this doesn't work here. But I think the end of the comment (from "So, when user clicks...") is not so useful, so you can probably just remove that part. Indeed the problem isn't that the focus is moved to the button, the problem is that the menu is mounted at load time and is hidden, so it's not the right way to do the focus.